Ghidra/Processors/CP1600/data/languages/CP1600.slaspec

```diff
@@ -0,0 +1,651 @@
+
+:HLT is opcode0_9=0x0000 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	goto inst_start;
+}
+
+:INCR reg0_2 is opcode3_9=0x0001 & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	reg0_2 = reg0_2 + 1;
+	resultFlags(reg0_2);
+}
+
+:J jmpdest16 is opcode0_9=0x0004 ; jump_type=0 & reg24_25=R7 & jmpdest16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	goto jmpdest16;
+}
+
+:JD jmpdest16 is opcode0_9=0x0004 ; jump_type=2 & reg24_25=R7 & jmpdest16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	I = 0;
+	goto jmpdest16;
+}
+
+:JE jmpdest16 is opcode0_9=0x0004 ; jump_type=1 & reg24_25=R7 & jmpdest16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	I = 1;
+	goto jmpdest16;
+}
+
+:JSR reg24_25, jmpdest16 is opcode0_9=0x0004 ; jump_type=0 & reg24_25 & jmpdest16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	reg24_25 = inst_next;
+	call jmpdest16;
+}
+
+:JSRD reg24_25, jmpdest16 is opcode0_9=0x0004 ; jump_type=2 & reg24_25 & jmpdest16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	I = 0;
+	reg24_25 = inst_next;
+	call jmpdest16;
+}
+
+:JSRE reg24_25, jmpdest16 is opcode0_9=0x0004 ; jump_type=1 & reg24_25 & jmpdest16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	I = 1;
+	reg24_25 = inst_next;
+	call jmpdest16;
+}
+
+:MOVR reg3_5, reg0_2 is opcode6_9=0x0002 & reg0_2 & reg3_5 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	reg0_2 = reg3_5;
+	resultFlags(reg0_2);
+}
+
+:MVI addr16, reg0_2 is opcode3_9=0x0050 & reg0_2 ; addr16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local ptr:2 = addr16;
+	reg0_2 = *(*:2 ptr);
+}
+
+:MVO reg0_2, addr16 is opcode3_9=0x0048 & reg0_2 ; addr16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local ptr:2 = addr16;
+	*ptr = reg0_2;
+}
+
+:MVO@ reg0_2, reg3_5 is opcode6_9=0x0009 & reg0_2 & reg3_5 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local ptr:2 = reg3_5; 
+	*ptr = reg0_2;
+	if ((reg3_5 != R4) && (reg3_5 != R5) && (reg3_5 != R6)) goto inst_next;
+	reg3_5 = reg3_5 + 1;
+}
+
+:MVOI reg0_2 is opcode3_9=0x004F & reg0_2 ; imm16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local tmp:2 = inst_start + 2;
+	*tmp = reg0_2;
+}
+
+:NEGR reg0_2 is opcode3_9=0x0004 & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local tmp = reg0_2 ^ 0xFFFF;
+	local tmpC = carry(tmp, 1);
+	local tmpO = scarry(tmp, 1);
+	reg0_2 = -reg0_2;
+	C = tmpC;
+	O = tmpO;
+	resultFlags(reg0_2);
+}
+
+:NOP is opcode1_9=0x001A [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+}
+
+:NOPP is opcode6_9=0x0008 & branch_external=0 & branch_condition=8 ; imm16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+}
+
+:RSWD reg0_2 is opcode3_9=0x0007 & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	C = (reg0_2 & 0b00001000) != 0;
+	O = (reg0_2 & 0b00010000) != 0;
+	Z = (reg0_2 & 0b00100000) != 0;
+	S = (reg0_2 & 0b01000000) != 0;
+}
+
+:SDBD is opcode0_9=0x0001 [ doublebyte=1; globalset(inst_next, doublebyte); ] {
```

Comment:
A better solution to the use of [ doublebyte=0; globalset(inst_next, doublebyte); ] in every instruction would be to set the context variable doublebyte to noflow. Then you only need to set it here and ignore the clearing of it for every other instruction. See LRSet in the ARM.sinc file.

Reply:
That's indeed better, and for some reason I also missed that in the documentation...  Anyway, this should be fixed now.